### ISSUE TITLE
$ is sometimes necessary in (( … ))

### ISSUE THREAD

**Issue by mina86** [koalaman/shellcheck#684]
Superfluous warning when `${var-default}` is used inside of `$(( … ))` as seen in:
```
$ cat a.sh
test=$(( ${COLUMNS-80} - 10))
echo $test
$ shellcheck a.sh

In a.sh line 1:
test=$(( ${COLUMNS-80} - 10))
         ^-- SC2004: $ on variables in (( )) is unnecessary.
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.3.3
license: GNU Affero General Public License, version 3
website: http://www.shellcheck.net
$
```

**Comment by brother** [comment]
The warning is indeed misleading. Having the colon present makes it behave as expected. 

> ${parameter-default} and ${parameter:-default} are almost equivalent. The extra : makes a difference only when parameter has been declared, but is null.

**Comment by koalaman** [comment]
Indeed. Fixed in 6a71ff6f
